packages/dashboard/src/components/widgets/primitives/input/hooks/useUpdateAssetPropertyValue.ts

```diff
@@ -0,0 +1,45 @@
+import { BatchPutAssetPropertyValueCommand, BatchPutAssetPropertyValueResponse } from '@aws-sdk/client-iotsitewise';
+import { useContext } from 'react';
+import { ClientContext } from '~/components/dashboard/clientContext';
+
+const useUpdateAssetPropertyValue = () => {
+  const client = useContext(ClientContext);
+
+  return ({
+    assetId,
+    propertyId,
+    propertyAlias,
+    value,
+  }: {
+    assetId: string;
+    propertyId: string;
+    propertyAlias?: string;
+    value: string;
+  }): Promise<BatchPutAssetPropertyValueResponse | undefined> => {
+    if (!client) return Promise.resolve(undefined);
+    return client.send(
+      new BatchPutAssetPropertyValueCommand({
+        entries: [
+          {
+            entryId: 'inputWidget',
```

Comment:
I see your point. This is only used once though and I think creating a variable for this is a bit overkill.

Reply:
Good point. I chose not create a variable because it’s only used once and not relevent to the code but required by the api. I’ll add a variable for this with comments.